examples/summarization/inference_compare.py

```diff
@@ -187,39 +188,42 @@ def reward_model(accelerator, model, inputs):
 
         with torch.no_grad():
             generated_sequences = generate_from_prompt(
-                inputs["input_ids_prompt"].to(accelerator.device),
-                inputs["attention_mask_prompt"].to(accelerator.device),
+                inputs["input_ids"].to(accelerator.device),
+                inputs["attention_mask"].to(accelerator.device),
                 model,
                 script_args,
             )
 
-            #TODO 
-            #create input ids that are right-padded with tokenizer.pad_token_id
-            #create corresponding attention mask
-            #create labels corresponding to the generated tokens (not including prompt) 
-            #should have -100 (label_pad_token) everywhere else
-            reward_model_input = {
-                "input_ids": None,
-                "attention_mask": generated_attention_mask,
-                "labels": None,
+            generated_texts = tokenizer.batch_decode(generated_sequences, skip_special_tokens=True)
+            tokenizer.padding_side = "right"
+            generated_encodings = tokenizer(
+                generated_texts, padding=True, truncation=True, max_length=script_args.seq_length, return_tensors="pt"
+            )
+
+            generated_labels = generated_encodings["input_ids"].clone()
+            generated_labels[generated_labels == tokenizer.pad_token_id] = -100
+            prompt_lens = inputs["attention_mask"].sum(-1)
+            for i, prompt_len in enumerate(prompt_lens):
+                generated_labels[i * 2 : i * 2 + 2, :prompt_len] = -100
```

Comment:
This could be an issue since the `input` encoding won't necessarily correspond to the `generated_encoding`. E.g. if `TLDR: ` gets encoded as two token `TLDR:` and ` ` in the `input` but then `TLDR: hello` in `generated_encodings` also gets encoded as two tokens `TLDR:` and ` hello` then you have an issue since you'd be making ` hello` a padding token. 

If you check and this isn't the case, that's fine. You could also just skip the detokenization, tokenization and then it isn't an issue

Reply:
`prompt_len - 1` should do the trick. input prompt will be something like `[... , 4976,    27, 209]` and generations will be `[4976,    27, 50276, 30592, ....]`.

Reply:
makes sense, you can compute log ps for prompt as well